mod.ts

```diff
@@ -99,8 +99,13 @@ export class App {
         Object.assign(params, parseURLSearchParams(search));
       }
     } else {
-      const decodedBody = decode(await readAll(req.body)); // FIXME: this line is should be refactored using Deno.Reader
-      const contentType = req.headers.get('content-type');
+      const rawContentType = req.headers.get('content-type') || "application/octet-stream";
+      const [contentType, ...typeParamsArray] = rawContentType.split(';')
+      const typeParams = Object.assign({}, ...typeParamsArray.map(params => params.split('=')))
+
+      const decoder = new TextDecoder(typeParams['charset'] || "utf-8")
```

Comment:
To get `charset` in this form, keys must be transformed using function like `downcase` in lodash.
I just add TODO comment for this ;)
```suggestion
      const decoder = new TextDecoder(typeParams['charset'] || "utf-8"); // TODO: downcase `charset` key
```